subprojects/core/src/main/java/org/gradle/api/internal/artifacts/CachingDependencyResolveContext.java

```diff
@@ -60,7 +59,9 @@ public FileCollection resolve() {
         }
     }
 
-    @Override
+    /**
+     * Accepts either a {@link ResolvableDependency} or {@link org.gradle.api.file.FileCollection}
```

Comment:
🤔 This class is internal, can you just split this into 2 overloaded methods that are strongly-typed?

Reply:
Yes likely but I wanted to avoid messing with this. This class is deprecated and will be removed